data/basal/cleanup.go

```diff
@@ -0,0 +1,65 @@
+package basal
+
+import (
+	"fmt"
+	"sort"
+	"time"
+
+	"github.com/mdblp/tide-whisperer-v2/v2/schema"
+	schemaV1 "github.com/tidepool-org/tide-whisperer/schema"
+)
+
+func CleanUpBasals(basals []schema.BasalBucket, loopModes []schemaV1.LoopModeEvent) []schema.BasalBucket {
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches to further communicate your intent.  -->
**🎯 suggestion (non-blocking):** ‏ I can see three functionalities here : the filling when basal are not continuous, the overlap handling and finally the join between loopModes and basal deliveryType value. I would have separate this in distinct functions (for testing purpose mainly). I can see that the for loop for the overlap is not tested by the way (this for loop is a recursive function, no ?)